geode-core/src/main/java/org/apache/geode/management/internal/ManagementCacheListener.java

```diff
@@ -33,20 +35,18 @@
 
   private static final Logger logger = LogService.getLogger();
 
-  private MBeanProxyFactory proxyHelper;
+  private final MBeanProxyFactory proxyHelper;
 
-  private volatile boolean readyForEvents;
+  private final CountDownLatch readyForEvents;
```

Comment:
The latch should probably be an instance of StoppableCountDownLatch so that it will abort (by throwing) if the cache is closed while threads are waiting on the latch. Otherwise, this has the potential to hang.